packages/artifact/src/internal/client.ts

```diff
@@ -225,4 +244,41 @@ If the error persists, please check whether Actions and API requests are operati
       throw error
     }
   }
+
+  async deleteArtifact(
+    artifactName: string,
+    options?: FindOptions
+  ): Promise<DeleteArtifactResponse> {
+    try {
+      if (isGhes()) {
+        throw new GHESNotSupportedError()
+      }
+
+      if (options?.findBy) {
```

Comment:
I'm imagining an edge cases where someone provides `options?.findBy` but the `repositoryOwner`, `repositoryName` and `workflowRunID` all match the current run so it should technically go and try to use the internal delete method. Do you think we need this bit of logic here?

Reply:
At least for the folks using typescript, you _must_ provide a token when using `findBy` because of it's typing: https://github.com/actions/toolkit/blob/1b5a6e26f42769414fe2302e6a6e8b5ac7bf9600/packages/artifact/src/internal/shared/interfaces.ts#L128-L135

I have this same `if (options?.findBy)` check in other methods, honestly I think it's good enough. If we were to change it, I would also want to do the same for the others as well.